src/models/single_publisher_impression_model.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base class for modeling single pub reach from impression counts.
+
+A single publisher impression model is a function that maps impression
+counts to number of users reached.  To fit the model, one or more points
+on the reach curve are provided.
+"""
+
+from typing import NamedTuple
+
+
+class ReachPoint(NamedTuple):
+  """A single point on a reach curve.
+
+  impressions:  The number of impressions that were shown.
+  reach_at_frequency: A tuple of values representing the number of people
+    reached at each frequency.  reach_at_frequency[i] is the number of
+    people who were reached AT LEAST i+1 times.
+  """
+  impressions: int
+  reach_at_frequency: list[int]
+
+
+class ImpressionsToReachModel:
```

Comment:
An alternative naming can be : 

For Single Publisher impression to reach -> ReachCurve
For Multi Publisher impression to reach -> ReachSurface

For Single Publisher spend to reach -> doesn't exist, can be achieved with ReachBySpendSurface or ReachBySpendCurve

For Multi Publisher spend to reach ->  ReachBySpendSurface 

We used this terminology in discussions internally. Wdyt?

Reply:
I think you are right here.  I changed the name of the class to ReachCurve and the name of the module to reach_by_impression_model.  Let me know if you like this better.

Reply:
Sounds good.